ts/src/profilers/heap-profiler.ts

```diff
@@ -15,41 +15,52 @@
  */
 
 import {perftools} from '../../../proto/profile';
+import {defaultConfig} from '../config';
+
 import {serializeHeapProfile} from './profile-serializer';
 
 const profiler = require('bindings')('sampling_heap_profiler');
 
-export class HeapProfiler {
-  private enabled = false;
-
-  /**
-   * @param intervalBytes - average bytes between samples.
-   * @param stackDepth - upper limit on number of frames in stack for sample.
-   */
-  constructor(private intervalBytes: number, private stackDepth: number) {
-    this.enable();
-  }
+let enabled = false;
+let heapIntervalBytes = defaultConfig.heapIntervalBytes;
+let heapStackDepth = defaultConfig.heapMaxStackDepth;
 
-  /**
-   * Collects a heap profile when heapProfiler is enabled. Otherwise throws
-   * an error.
-   */
-  profile(): perftools.profiles.IProfile {
-    if (!this.enabled) {
-      throw new Error('Heap profiler is not enabled.');
-    }
-    const result = profiler.getAllocationProfile();
-    const startTimeNanos = Date.now() * 1000 * 1000;
-    return serializeHeapProfile(result, startTimeNanos, this.intervalBytes);
+/*
+ * Collects a heap profile when heapProfiler is enabled. Otherwise throws
+ * an error.
+ */
+export function profile(): perftools.profiles.IProfile {
+  if (!enabled) {
+    throw new Error('Heap profiler is not enabled.');
   }
+  const result = profiler.getAllocationProfile();
+  const startTimeNanos = Date.now() * 1000 * 1000;
```

Comment:
It would be slightly better to record the start time before calling getAllocationProfile maybe?

Reply:
done.